frontend/types/api/notifications.ts

```diff
@@ -0,0 +1,176 @@
+  node_address: Hash;
+}
+export type InternalGetNotificationRocketPool = ApiPagingResponse<NotificationRocketPoolTableRow>;
+/**
+ * ------------------------------------------------------------
+ * Networks Table
+ */
+export interface NotificationNetworksTableRow {
+  chain_id: number /* uint64 */;
+  timestamp: number /* uint64 */;
+  event_type: string;
+  alert_value: string /* decimal.Decimal */; // wei string for gas alerts, otherwise percentage (0-1) for participation rate
+}
+export type InternalGetNotificationNetworks = ApiPagingResponse<NotificationNetworksTableRow>;
+/**
+ * ------------------------------------------------------------
+ * Notification Settings
+ */
+export interface NotificationEventsNetwork {
+  gas_above: string /* decimal.Decimal */;
+  gas_below: string /* decimal.Decimal */;
+  participation_rate: number /* float64 */;
+}
+export interface NotificationSettingsNetwork {
+  chain_id: number /* uint64 */;
+  subscribed_events: NotificationEventsNetwork;
+}
+export type InternalPutNotificationSettingsNetworksResponse = ApiDataResponse<NotificationSettingsNetwork>;
+export interface NotificationSettingsPairedDevice {
+  id: string;
+  name?: string;
+  enable_notifications: boolean;
+}
+export type InternalPutNotificationSettingsPairedDevicesResponse = ApiDataResponse<NotificationSettingsNetwork>;
+export interface NotificationEventsGeneral {
+  machines_offline: boolean;
+  machines_storage_usage: number /* float64 */;
+  machines_cpu_usage: number /* float64 */;
+  machines_memory_usage: number /* float64 */;
+  clients: string[];
+  clients_rocket_pool_smart_node: boolean;
+  clients_mev_boost: boolean;
+  rocket_pool_new_reward_round: boolean;
+  rocket_pool_max_collateral: number /* float64 */;
+  rocket_pool_min_collateral: number /* float64 */;
+}
+export interface NotificationSettings {
+  do_not_disturb_timestamp: number /* uint64 */;
+  enable_email: boolean;
+  enable_push: boolean;
+  subscribed_events: NotificationEventsGeneral;
+  networks: NotificationSettingsNetwork[];
+  paired_devices: NotificationSettingsPairedDevice[];
+}
+export type InternalGetNotificationSettingsResponse = ApiDataResponse<NotificationSettings>;
+export interface NotificationEventsValidatorDashboard {
+  validator_offline: boolean;
+  group_offline: number /* float64 */;
+  attestations_missed: boolean;
+  block_proposal: boolean;
+  upcoming_block_proposal: boolean;
+  sync: boolean;
+  withdrawal_processed: boolean;
+  slashed: boolean;
+  realtime_mode: boolean;
+}
+export interface NotificationSettingsValidatorDashboard {
+  webhook_url: string;
+  webhook_discord: boolean;
+  real_time_mode: boolean;
+  subscribed_events: NotificationEventsValidatorDashboard;
+}
+export type InternalPutNotificationSettingsValidatorDashboardResponse = ApiDataResponse<NotificationSettingsValidatorDashboard>;
+export interface NotificationEventsAccountDashboard {
```

Comment:
Is this the part that tells us how to send the API the preferences of the user in this dialog?
![image](https://github.com/gobitfly/beaconchain/assets/150015231/d7e83feb-502c-4e1f-954d-e81408b8dd33)
The user can select networks and can ask us to ignore spam transactions.
I do not see those fields in this json :)

Reply:
Network selection is in `NotificationSettingsAccountDashboard`, this type is supposed to only contain subscribable events. But @MauserBitfly would prefer flat types, so I will add all fields from the events types to their respective settings type.